File: GeneratedEntityReaderClasses/Components/TelekinesisComponent.cs
namespace GeneratedNoitaClasses;
public class TelekinesisComponent : NoitaComponentBase,  iNoitaType<TelekinesisComponent>{
	public NoitaBool Deleted { get; set; } = new();
	public NoitaBool Enabled { get; set; } = new();
	public NoitaString Tags { get; set; } = new();
	public NoitaUInt min_size { get; set; } = new();
	public NoitaUInt max_size { get; set; } = new();
	public NoitaFloat radius { get; set; } = new();
	public NoitaFloat throw_speed { get; set; } = new();
	public NoitaFloat target_distance { get; set; } = new();
	public NoitaBool kick_to_use { get; set; } = new();
	public NoitaInt mState { get; set; } = new();
	public NoitaUInt64 mBodyID { get; set; } = new();
	public NoitaFloat mStartBodyMaxExtent { get; set; } = new();
	public NoitaFloat mStartAimAngle { get; set; } = new();
	public NoitaFloat mStartBodyAngle { get; set; } = new();
	public NoitaFloat mStartBodyDistance { get; set; } = new();
	public NoitaFloat mStartTime { get; set; } = new();
	public NoitaFloat mMinBodyDistance { get; set; } = new();
	public NoitaBool mInteract { get; set; } = new();

    public void Read(NoitaStream s){
		Deleted.Read(s);
		Enabled.Read(s);
		Tags.Read(s);
		min_size.Read(s);
		max_size.Read(s);
		radius.Read(s);
		throw_speed.Read(s);
		target_distance.Read(s);
		kick_to_use.Read(s);
		mState.Read(s);
		mBodyID.Read(s);
		mStartBodyMaxExtent.Read(s);
		mStartAimAngle.Read(s);
		mStartBodyAngle.Read(s);
		mStartBodyDistance.Read(s);
		mStartTime.Read(s);
		mMinBodyDistance.Read(s);
		mInteract.Read(s);

    }
    public void Write(NoitaStream s){
		Deleted.Write(s);
		Enabled.Write(s);
		Tags.Write(s);
		min_size.Write(s);
		max_size.Write(s);
		radius.Write(s);
		throw_speed.Write(s);
		target_distance.Write(s);
		kick_to_use.Write(s);
		mState.Write(s);
		mBodyID.Write(s);
		mStartBodyMaxExtent.Write(s);
		mStartAimAngle.Write(s);
		mStartBodyAngle.Write(s);
		mStartBodyDistance.Write(s);
		mStartTime.Write(s);
		mMinBodyDistance.Write(s);
		mInteract.Write(s);

    }
}

File: GeneratedEntityReaderClassesPython/GeneratedClasses/Components/TelekinesisComponent.py
from dataclasses import dataclass, field
from BasicNoitaTypes import NoitaBool
from BasicNoitaTypes import NoitaString
from BasicNoitaTypes import NoitaUInt
from BasicNoitaTypes import NoitaFloat
from BasicNoitaTypes import NoitaInt
from BasicNoitaTypes import NoitaUInt64

from NoitaStream import NoitaStream
@dataclass
class TelekinesisComponent:
    Deleted:NoitaBool = field(default_factory=NoitaBool)
    Enabled:NoitaBool = field(default_factory=NoitaBool)
    Tags:NoitaString = field(default_factory=NoitaString)
    min_size:NoitaUInt = field(default_factory=NoitaUInt)
    max_size:NoitaUInt = field(default_factory=NoitaUInt)
    radius:NoitaFloat = field(default_factory=NoitaFloat)
    throw_speed:NoitaFloat = field(default_factory=NoitaFloat)
    target_distance:NoitaFloat = field(default_factory=NoitaFloat)
    kick_to_use:NoitaBool = field(default_factory=NoitaBool)
    mState:NoitaInt = field(default_factory=NoitaInt)
    mBodyID:NoitaUInt64 = field(default_factory=NoitaUInt64)
    mStartBodyMaxExtent:NoitaFloat = field(default_factory=NoitaFloat)
    mStartAimAngle:NoitaFloat = field(default_factory=NoitaFloat)
    mStartBodyAngle:NoitaFloat = field(default_factory=NoitaFloat)
    mStartBodyDistance:NoitaFloat = field(default_factory=NoitaFloat)
    mStartTime:NoitaFloat = field(default_factory=NoitaFloat)
    mMinBodyDistance:NoitaFloat = field(default_factory=NoitaFloat)
    mInteract:NoitaBool = field(default_factory=NoitaBool)
    
    def Read(self, s:NoitaStream):
        self.Deleted.Read(s)
        self.Enabled.Read(s)
        self.Tags.Read(s)
        self.min_size.Read(s)
        self.max_size.Read(s)
        self.radius.Read(s)
        self.throw_speed.Read(s)
        self.target_distance.Read(s)
        self.kick_to_use.Read(s)
        self.mState.Read(s)
        self.mBodyID.Read(s)
        self.mStartBodyMaxExtent.Read(s)
        self.mStartAimAngle.Read(s)
        self.mStartBodyAngle.Read(s)
        self.mStartBodyDistance.Read(s)
        self.mStartTime.Read(s)
        self.mMinBodyDistance.Read(s)
        self.mInteract.Read(s)
    
    def Write(self, s:NoitaStream):
        self.Deleted.Write(s)
        self.Enabled.Write(s)
        self.Tags.Write(s)
        self.min_size.Write(s)
        self.max_size.Write(s)
        self.radius.Write(s)
        self.throw_speed.Write(s)
        self.target_distance.Write(s)
        self.kick_to_use.Write(s)
        self.mState.Write(s)
        self.mBodyID.Write(s)
        self.mStartBodyMaxExtent.Write(s)
        self.mStartAimAngle.Write(s)
        self.mStartBodyAngle.Write(s)
        self.mStartBodyDistance.Write(s)
        self.mStartTime.Write(s)
        self.mMinBodyDistance.Write(s)
        self.mInteract.Write(s)

